README.md

```diff
@@ -801,6 +801,33 @@ ctx.run(query[Book])
 // SELECT id, notes, pages, history FROM Book
 ```
 
+### User-Defined Types
+
+The cassandra context provides encoding of UDT (user-defined types).
+```scala
+import io.getquill.context.cassandra.Udt
+
+case class Name(firstName: String, lastName: String) extends Udt
+```
+
+It's possible to override UDT name or columns using `udtMeta`:
+```scala
+implicit val nameMeta = udtMeta[Name]("my_name", _.firstName -> "first")
+```
+
+To translate UDT name/columns from case classes, the cassandra context first tries to use definitions from `udtMeta`
+and only if those could not be found, then translates them using `NamingStrategy`.
+
+It's also possible to specify UDT keyspace using `udtMeta`:
+
+```scala
+implicit val nameMeta = udtMeta[Name]("keyspace2.my_name", _.lastName -> "last_name")
+```
+
+When keyspace is not explicitly set then context searches for UDT within logged keypace.
+Since it's possible to create cassandra context with `null` keyspace, e.g. session is not bound to any keyspace,
+then context tries to find UDT metadata within all cluster.
```

Comment:
@optician Now the way of resolving UDT metadata is more obvious and deterministic.

To answer your question regarding compile time resolving, not it's no possible since it needs datastax cluster to be created and connected to retrieve metadata

Reply:
My mistake. Anyway runtime error is possible for dynamic queries.

I have no long time experience with cassandra and quill and maybe my reasoning is unsound. 

Let's keyspace of `CassandraContext` is `null`. To send a correct query we should specify keyspaces for all database objects (tables, udt). If something is wrong then cassandra answers with error. 
What I don't understand is a need of cluster meta information to use keyspace in query. If metadata is used to check correctness then 
* It's possible to perform some checks before dispatch of query. Looks like a good feature because it decreases useless load and waste of resources (when price of analysis is low). Fast fail approach.
* It raises a problem of metadata actuality. How often information updates? Does it perform before each query or at start of application or maybe within certain period of background process?

Also a try to guess keyspace by search in metadata is error prone:
* Situation when two UDT with the same name exist. Two ways to solve - cast a dice or check keyspace equality.
* Human fault. Only one UDT exists but it exists in separate keyspace. Maybe keyspace is wrong or maybe udt is not created in right keyspace. 

My suggest is to exclude uncertain situations. Keyspace of UDT should be defined by user or be taken the same as for table. Last case maybe broken for joins from different keyspace (I don't know is it possible at all).

Reply:
@optician Well, I have not explained why would I need UDT metadata. For select queries (mostly decoders) I'm just pulling this from result row. But for insert/updates (mostly encoders), in order to bind UDT, I need to create datastax `UDTValue`, which I could do by using UDT metadata, which I can pull form keyspace metadata. Please look at code generated above ` ctx.udtValueO` is used only for encoders.

I have updated README so it should look more obvious, please say if not.

And finally, there's will be not uncertain situation. Cassandra won't let you bind wrong UDT to specific query, since it would expect another :)